control/control_security_test.go

```diff
@@ -0,0 +1,526 @@
+// +build medium
+
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2017 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package control
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/intelsdi-x/gomit"
+	. "github.com/smartystreets/goconvey/convey"
+
+	"github.com/intelsdi-x/snap/control/fixtures"
+	"github.com/intelsdi-x/snap/control/plugin"
+	"github.com/intelsdi-x/snap/control/plugin/client"
+	"github.com/intelsdi-x/snap/core"
+	"github.com/intelsdi-x/snap/core/cdata"
+	"github.com/intelsdi-x/snap/core/ctypes"
+	"github.com/intelsdi-x/snap/plugin/helper"
+)
+
+type MockEmitter struct{}
+
+const (
+	tlsTestCAFn  = "snaptest-CA"
+	tlsTestSrvFn = "snaptest-srv"
+	tlsTestCliFn = "snaptest-cli"
+)
+
+var tlsTestCA, tlsTestSrv, tlsTestCli string
+
+var testFilesToRemove []string
+
+func (memitter *MockEmitter) Emit(gomit.EventBody) (int, error) { return 0, nil }
+
+type configTLSMock Config
+
+func (m *configTLSMock) export() *Config {
+	return (*Config)(m)
+}
+
+func (m *configTLSMock) setTLSCertPath(certPath string) *configTLSMock {
+	m.TLSCertPath = certPath
+	return m
+}
+
+func (m *configTLSMock) setTLSKeyPath(keyPath string) *configTLSMock {
+	m.TLSKeyPath = keyPath
+	return m
+}
+
+func TestMain(m *testing.M) {
+	setUpTestMain()
+	retCode := m.Run()
+	tearDownTestMain()
+	os.Exit(retCode)
+}
+
+func TestSecureCollector(t *testing.T) {
+	log.SetLevel(log.DebugLevel)
+	Convey("Having a secure collector", t, func() {
+		var ap *availablePlugin
+		Convey("framework should establish secure connection", func() {
+			security := client.SecurityTLSExtended(tlsTestCli+fixtures.TestCrtFileExt, tlsTestCli+fixtures.TestKeyFileExt, client.SecureClient, []string{tlsTestCA + fixtures.TestCrtFileExt})
```

Comment:
Sadly, that's standard formatting enforced by `gofmt`. I would prefer the format with spaces too, and that's how it looked like before `gofmt` vandalized my work.